File: mods/java/common/src/main/java/com/alaydriem/bedrockvoicechat/dto/Orientation.java
package com.alaydriem.bedrockvoicechat.dto;

import lombok.Getter;

/**
 * Player orientation (yaw and pitch).
 */
@Getter
public class Orientation {
    /**
     * Yaw (horizontal rotation) - stored as 'x' for API compatibility.
     */
    private final float x;

    /**
     * Pitch (vertical rotation) - stored as 'y' for API compatibility.
     */
    private final float y;

    public Orientation(float yaw, float pitch) {
        this.x = yaw;
        this.y = pitch;
    }
}

File: mods/java/common/src/main/kotlin/com/alaydriem/bedrockvoicechat/dto/Orientation.kt
package com.alaydriem.bedrockvoicechat.dto

/**
 * Player orientation.
 * x = pitch (up/down, ±90°), y = yaw (facing direction, ±180°)
 */
data class Orientation(
    val x: Float,
    val y: Float
) {
    companion object {
        fun fromYawPitch(yaw: Float, pitch: Float) = Orientation(x = pitch, y = yaw)
    }
}
